library/cwm/src/lib/cwm/replace_point.rb

```diff
@@ -0,0 +1,69 @@
+require "yast"
+require "cwm/custom_widget"
+
+Yast.import "UI"
+Yast.import "CWM"
+
+module CWM
+  # Placeholder widget that is used to replace content on demand.
+  # The most important method is {#replace} which allows switching content
+  class ReplacePoint < CustomWidget
+    # @param id [Object] id of widget. Needed to redefine only if more than one
+    # placeholder needed to be in dialog. Parameter type is limited by component
+    # system
+    # @param widget [CWM::AbstractWidget] initial widget in placeholder
+    def initialize(id: "_placeholder", widget: Empty.new("_initial_placeholder"))
+      self.handle_all_events = true
+      self.widget_id = id
+      @widget = widget
+    end
+
+    # @return [UITerm]
+    def contents
+      ReplacePoint(Id(widget_id), Empty(Id("____CWM_empty")))
```

Comment:
it make sense, this will also simplify init, as init will be done as part of CWM itself.